### PR TITLE
[SMALLFIX] Fixing the vagrant/deploy/{create,destroy} scripts.

### PR DESCRIPTION
**Pull request by jsimsa**
This PR makes sure that vagrant/deploy/{create,destroy} only invoke the vagrant/deploy/bin/is_aws_spot.py script for the aws provider. The create script takes the provider as an argument so that's easy. The destroy script does not, so we instead check if the vagrant/deploy/conf/ec2.yml file exists as a heuristic.

This PR also cleans up the style of the create and destroy scripts, using [[ ]] over [ ] and $() over `` consistently.

@uronce-cc

### CODE REVIEW

File: deploy/vagrant/create [hunk 1]
```diff
@@ -87,11 +87,10 @@ done
 echo "TachyonMaster" >> files/workers
 
 # create virtual machine without provisioning
-[[ "$PROVIDER" == "aws" ]] && sgid=`python bin/init_aws.py | tail -n 1`
+[[ "$PROVIDER" == "aws" ]] && sgid=$(python bin/init_aws.py | tail -n 1)
 # pass security group id to vagrant up
 export AWS_SECURITY_GROUP_ID_TACH=$sgid
-IS_SPOT=`python bin/is_aws_spot.py`
-if [[ "$PROVIDER" == "aws" && "$IS_SPOT" == 0 ]]; then
+if [[ "$PROVIDER" == "aws" && "$IS_SPOT" == "$(python bin/is_aws_spot.py)" ]]; then
```
Comment (miasantreble) [inline deploy/vagrant/create]: where did this $IS_SPOT come from?
I assume you meant `&& "0" == "$(python bin/is_aws_spot.py)"`?

Comment (jsimsa) [inline deploy/vagrant/create]: Good catch. Done.

Comment (uronce-cc) [inline deploy/vagrant/create]: +1

**Comment by uronce-cc** [comment]
+1 LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**